### DISCUSSION TITLE
How to change sdxl styles path? And how to stop default styles from being downloaded?

### DISCUSSION THREAD

**Discussion by TMFedd** [lllyasviel/Fooocus#2910] [Q&A] [answered]
I changed the paths for everything else on my config.txt file, but I can't figure out the correct property to change the path of my styles. I've tried path_styles and it didn't work.

I'm also tired of deleting the default styles, they keep downloading themselves again every once in a while, is there a way to disable that?

**Comment by mashb1t** [comment, marked as answer]
You can't configure the location of styles, they're hard-coded in https://github.com/lllyasviel/Fooocus/blob/6308fb8b54f62e61711aa57b086b30466ebbb857/modules/sdxl_styles.py#L10, but you can always change the files to load in styles https://github.com/lllyasviel/Fooocus/blob/6308fb8b54f62e61711aa57b086b30466ebbb857/modules/sdxl_styles.py#L30-L35

**Reply by TMFedd** [reply]
Well that sucks, but it is what it is.

I'm not sure what the point of that for loop is, adding new jsons to the default sdxl_styles folder already loads them in automatically, or am I missing something?

I'm still curious though, what causes the old sdxl styles to be redownloaded? Similar things happen when I manually change my .py/.css files, they eventually get "reset", even though there's no new fooocus version...

**Reply by mashb1t** [reply]
Fooocus needs this loop as only allowed style files will be added, not all in the folder. This is a historically grown code partial, not implemented by me.

Concerning the reset I can't tell for sure but i assume you've run git reset or similar or Fooocus detects the files as missing and adds them again from the git index. If you wabt to have them pemanently removed you can run `git add .` and `git commit -m "personal changes"`, but keep in mind that Fooocus is then not able to update itself automatically on the next release.
